include/picongpu/fields/background/templates/twtsfast/BField.tpp

```diff
@@ -353,7 +361,10 @@ namespace picongpu
                  * (i.e. from a finite coordinate range) only. All these quantities have to be calculated
                  * in double precision.
                  */
-                float_64 const tanAlpha = (float_64(1.0) - beta_0 * math::cos(phi)) / (beta_0 * math::sin(phi));
+                float_64 sinPhiVal;
+                float_64 cosPhiVal;
+                math::sincos(phi, sinPhiVal, cosPhiVal);
+                float_64 const tanAlpha = (1.0 - beta_0 * cosPhiVal) / (beta_0 * sinPhiVal);
                 float_64 const tanFocalLine = math::tan(PI / float_64(2.0) - phi);
                 float_64 const deltaT
                     = wavelength_SI / SI::SPEED_OF_LIGHT_SI * (float_64(1.0) + tanAlpha / tanFocalLine);
```

Comment:
Please only `1.0`, could you please check for other places in this PR where you used the same pattern.  We cast only to types with variable precision or if we need to downcast.

Reply:
Done for all occurrences in the PR.